### PR TITLE
[https://jira.cms.gov/projects/WNMGDS-198/](WNMGDS-198) Update icon docs

### PR DESCRIPTION
**Pull request by phiden**
### Added
- Added `style/icons` 
- lists all icons available in sketch file (except m.gov specific ones) plus links to download the SVGs

### CODE REVIEW

**Comment by line47** [comment]
I'm so glad we will have an icons page soon! 🎉 

The build is failing at the moment due to the SCSS for Alert using info.svg and that file no longer exists as it has been renamed. https://github.com/CMSgov/design-system/blob/update-icon-docs/packages/core/src/components/Alert/Alert.scss#L32

There are also a few linting errors in the icons.scss file 
```
packages/core/src/base/icons.scss
  88:3  ✖  Expected height to come before width                        order/properties-alphabetical-order
  95:5  ✖  Expected background-repeat to come before background-size   order/properties-alphabetical-order
 101:5  ✖  Expected background-repeat to come before background-size   order/properties-alphabetical-order
 109:5  ✖  Expected background-repeat to come before background-size   order/properties-alphabetical-order
 115:5  ✖  Expected background-repeat to come before background-size   order/properties-alphabetical-order
 121:5  ✖  Expected background-repeat to come before background-size   order/properties-alphabetical-order
 128:5  ✖  Expected background-repeat to come before margin-top        order/properties-alphabetical-order
 134:5  ✖  Expected background-repeat to come before background-size   order/properties-alphabetical-order
 141:5  ✖  Expected background-repeat to come before background-size   order/properties-alphabetical-order
 148:5  ✖  Expected background-repeat to come before background-size   order/properties-alphabetical-order
 154:5  ✖  Expected background-repeat to come before background-size   order/properties-alphabetical-order
 160:5  ✖  Expected background-repeat to come before background-size   order/properties-alphabetical-order
 169:5  ✖  Expected background-repeat to come before margin-top        order/properties-alphabetical-order
 175:5  ✖  Expected background-repeat to come before background-size   order/properties-alphabetical-order
 184:5  ✖  Expected background-repeat to come before background-size   order/properties-alphabetical-order
 193:5  ✖  Expected background-repeat to come before background-size   order/properties-alphabetical-order
 201:5  ✖  Expected background-repeat to come before background-size   order/properties-alphabetical-order```

**Comment by phiden** [comment]
Drives me crazy that my linter doesn't pick any of that up. Fixing.

File: packages/core/src/base/icons.scss [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+
+@hide-markup
+
+These icons are used primarily as part of the Alert Box component, though the Checkmark can sometimes be found in other places, without the background.
+
+Markup:
+<% var icons = [
+  'information', 'success', 'success-fill', 'warning', 'error'
+]; -%>
+<div class="ds-l-container">
```
Comment (line47) [inline packages/core/src/base/icons.scss]: Wondering about changing this class from `ds-l-container` to `ds-u-padding--1` as it pulls the icons a bit closer to the left edge. 
### ds-u-padding--1 screenshot
Screen Shot 2019-07-09 at 10 03 53 AM

### ds-l-container screenshot
Screen Shot 2019-07-09 at 10 03 59 AM

Comment (phiden) [reply packages/core/src/base/icons.scss]: sure why not

**Comment by line47** [comment]
for the linting you can run `yarn lint` before you commit to catch those issues locally.

**Comment by phiden** [comment]
Why is that not a pre-commit hook?

**Comment by line47** [comment]
Looks like `lint-staged` [is set up](https://github.com/CMSgov/design-system/blob/update-icon-docs/package.json#L12) but only to lint JS/JSX, we can add CSS/SASS/SCSS to the pre-commit. https://github.com/CMSgov/design-system/blob/update-icon-docs/package.json#L110

I'll add a JIRA issue to add that to the pre-commit 👍

File: packages/core/src/base/icons.scss [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+@hide-markup
+
+These icons are used primarily as part of the Alert Box component, though the Checkmark can sometimes be found in other places, without the background.
+
+Markup:
+<% var icons = [
+  'information', 'success', 'success-fill', 'warning', 'error'
+]; -%>
+<div class="ds-l-container ds-u-padding--1 ">
```
Comment (line47) [inline packages/core/src/base/icons.scss]: We'll need to remove the `ds-l-container` class from these divs to get the icons pulled to the left like in the screenshots provided

File: packages/core/src/base/icons.scss [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+
+@hide-markup
+
+These icons are used primarily for interaction with the user interface.
+
+Markup:
+<% var icons = [
+  'close', 'arrow-up', 'arrow-down'
+]; -%>
+<div class="ds-l-container ds-u-padding--1">
```
Comment (line47) [inline packages/core/src/base/icons.scss]: We'll need to remove the `ds-l-container` class from these divs to get the icons pulled to the left like in the screenshots provided

File: packages/core/src/base/icons.scss [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+
+@hide-markup
+
+These icons are used to convey more information about a given block of text or interaction. They should be used sparingly.
+
+Markup:
+<% var icons = [
+  'star','star-filled', 'information-new', 'information-new-hover', 'PDF', 'image', 'download', 'next'
+]; -%>
+<div class="ds-l-container ds-u-padding--1">
```
Comment (line47) [inline packages/core/src/base/icons.scss]: We'll need to remove the `ds-l-container` class from these divs to get the icons pulled to the left like in the screenshots provided

**Pull request merged**